### PR TITLE
Add Ruby 3 support

### PR DESCRIPTION
**Pull request by Taeir**
Changes required to support ruby 3 are minimal: changing some arguments for YAML.safe_load.

Omniauth is updated to fix ruby 3 support, but I was not able to test whether the transferral of content from Stack Exchange is affected by the upgrade. Based on the changelog it should be fine though.

### CODE REVIEW

**Comment by cellio** [comment]
@ArtOfCode- this is still a draft so not adding reviewers yet, but heads up that this one looks big and will definitely need your attention later.

**Comment by Taeir** [comment]
> @ArtOfCode- this is still a draft so not adding reviewers yet, but heads up that this one looks big and will definitely need your attention later.

This is actually only about 20 lines of code. The big changes come from the rails upgrade on which this one depends. (#854)

**Comment by cellio** [comment]
> > @ArtOfCode- this is still a draft so not adding reviewers yet, but heads up that this one looks big and will definitely need your attention later.
> 
> This is actually only about 20 lines of code. The big changes come from the rails upgrade on which this one depends. (#854)

Oh, ok!  I saw the long list of commits and jumped to a wrong conclusion, sorry.

File: config/initializers/new_framework_defaults_6.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Embed purpose and expiry metadata inside signed and encrypted
+# cookies for increased security.
+#
+# This option is not backwards compatible with earlier Rails versions.
+Rails.application.config.action_dispatch.use_cookies_with_metadata = false
+
+# Specify cookies SameSite protection level: either :none, :lax, or :strict.
+#
+# This change is not backwards compatible with earlier Rails versions.
+Rails.application.config.action_dispatch.cookies_same_site_protection = nil
```
[the remark this answers is not in the document]
Comment (Taeir) [reply config/initializers/new_framework_defaults_6.rb]: No longer present in this PR (also not in the codebase).

**Pull request merged**